homeassistant/components/guardian/__init__.py

```diff
@@ -208,12 +215,19 @@ class GuardianEntity(CoordinatorEntity):
     """Define a base Guardian entity."""
 
     def __init__(  # pylint: disable=super-init-not-called
-        self, entry: ConfigEntry, kind: str, name: str, device_class: str, icon: str
+        self,
+        entry: ConfigEntry,
+        kind: str,
+        name: str,
+        device_class: str | None,
+        icon: str | None,
     ) -> None:
         """Initialize."""
         self._attr_device_class = device_class
         self._attr_device_info = {"manufacturer": "Elexa"}
-        self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: "Data provided by Elexa"}
+        self._attr_extra_state_attributes: MutableMapping[str, Any] = {
```

Comment:
Isn't typing infered?

Reply:
In this case, it would be inferred as dict, not MultableMapping.

Reply:
If I remove that explicit typing, code like this:

https://github.com/home-assistant/core/blob/edf42bab25ae2e3681f3d0e774b7eebdaaa87ce3/homeassistant/components/guardian/binary_sensor.py#L177-L183

...will throw this `mypy` error:

```
Item "None" of "Optional[MutableMapping[str, Any]]" has no attribute "update" 
```

Is it because the definition allows for a `None` value and I don't explicitly declare that my instance _isn't_ `None`?

https://github.com/home-assistant/core/blob/edf42bab25ae2e3681f3d0e774b7eebdaaa87ce3/homeassistant/helpers/entity.py#L230